actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -46,12 +51,21 @@ class ActionCable.Connection
     else
       @open()
 
+  getProtocol: ->
+    @webSocket?.protocol
+
+  isProtocolSupported: ->
+    @getProtocol() in protocols and @getProtocol() isnt unsupported_protocol
```

Comment:
> To top it off, I'd like to add the unsupported protocol back to `ActionCable::INTERNAL[:protocols]`, so we can think of it as a manifest of the negotiation order.

Sure I can do this. But on the server side, it does not matter which order they are given to the driver. It only matters that the client give the server the protocols with the unsupported protocol last. So I figured I would keep that ordering logic on the client side.